sycl/source/CMakeLists.txt

```diff
@@ -18,13 +28,23 @@ function(add_sycl_rt_library LIB_NAME)
   set_target_properties(${LIB_NAME} PROPERTIES LINKER_LANGUAGE CXX)
 
   if (MSVC)
-      target_compile_definitions(${LIB_NAME} PRIVATE __SYCL_BUILD_SYCL_DLL )
-      target_link_libraries(${LIB_NAME} PRIVATE shlwapi)
+    target_compile_definitions(${LIB_NAME} PRIVATE __SYCL_BUILD_SYCL_DLL )
+    target_link_libraries(${LIB_NAME} PRIVATE shlwapi)
+    if (SYCL_ENABLE_XPTI_TRACING)
+      target_link_libraries(${LIB_NAME} PRIVATE ${ARG_XPTI_LIB})
+    endif()
   else()
-      set(linker_script "${CMAKE_CURRENT_SOURCE_DIR}/ld-version-script.txt")
-      target_link_libraries(
-        ${LIB_NAME} PRIVATE "-Wl,--version-script=${linker_script}")
-      set_target_properties(${LIB_NAME} PROPERTIES LINK_DEPENDS ${linker_script})
+    set(linker_script "${CMAKE_CURRENT_SOURCE_DIR}/ld-version-script.txt")
+    target_link_libraries(
+      ${LIB_NAME} PRIVATE "-Wl,--version-script=${linker_script}")
+    set_target_properties(${LIB_NAME} PROPERTIES LINK_DEPENDS ${linker_script})
+    if (SYCL_ENABLE_XPTI_TRACING)
+      target_link_libraries(${LIB_NAME} PRIVATE ${ARG_XPTI_LIB} PRIVATE dl)
```

Comment:
You don't need to repeat `PRIVATE` twice. Anyway, you can restructure like this:

```
if (SYCL_ENABLE_XPTI_TRACING)
  target_link_libraries(${LIB_NAME} PRIVATE ${ARG_XPTI_LIB})
  target_compile_definitions(${LIB_NAME} PRIVATE XPTI_ENABLE_INSTRUMENTATION XPTI_STATIC_LIBRARY)
endif()
if (MSVC)
...
else()
...
  if (SYCL_ENABLE_XPTI_TRACING)
    target_link_libraries(${LIB_NAME} PRIVATE dl)
  endif()
endif()

Reply:
Done!